src/sec.rs

```diff
@@ -1456,13 +1532,21 @@ mod test {
         )
     }
 
-    // Tests the "normal flow" for a newly created saga: create + start.
-    #[tokio::test]
-    async fn test_saga_create_and_start_executes_saga() {
+    // We have a lot of tests which attempt to:
+    // - Inject some repeats
+    // - Inject some failures
+    // - Observe the count of "which nodes were called"
+    //
+    // This helper intends to reduce some of that boilerplate.
+    async fn saga_runner_helper(
+        repeat: Option<(NodeIndex, RepeatInjected)>,
```

Comment:
I find the tests using the helper somewhat hard to read. The `repeat` and `fail` parameters are tagged with comments and it's not immediately clear what the counts represent. What if instead we passed a struct to the saga runner helper so we could have named parameters? This would obviate the need for comments and on top of that give us a way to tie the counts to the named parameters. We could optionally add names to the counts by separating them into `node_redos` and `node_undos` in the struct.

Reply:
Done in 2c281c863a6293648b48f551597b0bdaea45eab7

Reply:
And also cbd43ff8380d000aa88bd784ee9d00c332812aa2

Reply:
Thanks @smklein. While, verbose, I think it makes the individual tests easier to read!